lib/contextual_logger.rb

```diff
@@ -5,94 +5,98 @@
 require_relative './contextual_logger/context/handler'
 
 module ContextualLogger
-  def self.new(logger)
-    logger.extend(self)
+  class << self
+    def new(logger)
+      logger.extend(LoggerMixin)
+    end
   end
 
+  module LoggerMixin
   def global_context=(context)
-    ContextualLogger::Context::Handler.new(context).set!
+    Context::Handler.new(context).set!
   end
 
   def with_context(context)
-    context_handler = ContextualLogger::Context::Handler.new(current_context_for_thread.deep_merge(context))
+    context_handler = Context::Handler.new(current_context_for_thread.deep_merge(context))
     context_handler.set!
     if block_given?
       yield
     else
-      context_handler
+      context_handler # JEB, is this actually useful? The set! and reset! wouldn't mean anything, so it would just be calling Context::Handler.new, right? -Colin
     end
   ensure
     context_handler.reset! if block_given?
   end
 
   def current_context_for_thread
-    ContextualLogger::Context::Handler.current_context
+    Context::Handler.current_context
   end
 
-  def format_message(severity, timestamp, progname, message, context)
-    message_with_context = message_with_context(context, message, severity, timestamp, progname)
-
-    if @formatter
-      @formatter.call(severity, timestamp, progname, message_with_context)
-    else
-      "#{message_with_context.to_json}\n"
-    end
+  def debug(message = nil, extra_context = {})
+    add_if_enabled(Logger::Severity::DEBUG, message || yield, extra_context: extra_context)
   end
 
-  def debug(progname = nil, **extra_context, &block)
-    add(Logger::Severity::DEBUG, nil, progname, extra_context, &block)
+  def info(message = nil, extra_context = {})
+    add_if_enabled(Logger::Severity::INFO, message || yield, extra_context: extra_context)
```

Comment:
This got a log simpler when I refactored to not pass the block and other unused args down.